pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -300,13 +300,13 @@ def on_train_epoch_end(
     ) -> None:
         """ Save a checkpoint at the end of the training epoch. """
         # as we advance one step at end of training, we use `global_step - 1` to avoid saving duplicates
-        trainer.train_loop.global_step -= 1
+        trainer.fit_loop.global_step -= 1
```

Comment:
wondering we do not have a test that would catch it?

Reply:
Yep, already here: https://github.com/PyTorchLightning/pytorch-lightning/blob/17f2ae5613c6213d8be011d05c05cf04d40b1d89/tests/deprecated_api/test_remove_1-6.py#L257

but of course, since it's getting removed we don't want to call it ourselves internally, otherwise we get a useless deprecation message.